concrete-core-fixture/src/generation/synthesizing/glwe_ciphertext.rs

```diff
@@ -63,4 +101,223 @@ mod backend_core {
             self.core_engine.destroy(entity).unwrap();
         }
     }
+
+    use crate::generation::synthesizing::SynthesizesGlweCiphertextView;
+    use concrete_core::prelude::{
+        GlweCiphertextConsumingRetrievalEngine, GlweCiphertextCreationEngine, GlweCiphertextEntity,
+        GlweCiphertextView32, GlweCiphertextView64,
+    };
+
+    impl<'a> SynthesizesGlweCiphertextView<Precision32, GlweCiphertextView32<'a>> for Maker {
+        fn synthesize_glwe_ciphertext_view(
+            &mut self,
+            prototype: &Self::GlweCiphertextProto,
+        ) -> GlweCiphertextView32<'a> {
+            let ciphertext = prototype.0.to_owned();
+            let polynomial_size = ciphertext.polynomial_size();
+            let container = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(ciphertext)
+                .unwrap();
+            self.core_engine
+                .create_glwe_ciphertext(container.leak() as &[u32], polynomial_size)
+                .unwrap()
+        }
+
+        fn unsynthesize_glwe_ciphertext_view(
+            &mut self,
+            entity: GlweCiphertextView32,
+        ) -> Self::GlweCiphertextProto {
+            let polynomial_size = entity.polynomial_size();
+            let slice = self
+                .core_engine
+                .consume_retrieve_glwe_ciphertext(entity)
+                .unwrap();
+            let ciphertext = self
+                .core_engine
+                .create_glwe_ciphertext(
+                    unsafe {
+                        Vec::from_raw_parts(slice.as_ptr() as *mut u32, slice.len(), slice.len())
+                    },
+                    polynomial_size,
+                )
+                .unwrap();
+            ProtoBinaryGlweCiphertext32(ciphertext)
+        }
+
+        fn destroy_glwe_ciphertext_view(
+            &mut self,
+            entity: GlweCiphertextView32,
+            drop_underlying_memory: bool,
```

Comment:
Here do we actually need to give this choice? I don't see a case where we wouldn't want to drop the underlying memory. @aPere3 what do you think?

Reply:
Well the thing for slices with the current context system for fixtures is that you cannot guarantee your memory to live long enough.

So what I did was leak the vecs to get slices. The thing is you may be able to get several immutable slices to the same memory, and if you put those i ciphertexts you may be in a situation of double/multiple frees if you destroy each time, so here I did not find a solution other than letting the developer tell us if the slice being unsynthesized is supposed to be the last slice to this memory so that we can free it

Reply:
We can do a quick call so I can explain the reasons here, we discussed them with Alex when he was advising me on the design but maybe not extensively with you

Reply:
this won't be a problem with the reworked design thanks to Alex, resolving